File: 268-missing-number/268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        
        int ideal = 0; 
        for(int i = 0; i <= nums.size(); i++)
        {
            ideal ^= i;
        }
        
        int given = 0; 
        
        for(int i = 0; i < nums.size(); i++)
        {
            given ^= nums[i];
        }
        
        
        return ideal ^ given; 
    }
};

File: 268-missing-number/268-missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        
        n = len(nums)
        
        
        expected_sum = 0
        for x in range(n + 1):
            expected_sum += x
            
        actual_sum = 0
        for _, x in enumerate(nums):
            
            actual_sum += x
        
        return expected_sum - actual_sum
            
        
        